src/app/api/v1/service-providers/sign-up/route.ts

```diff
@@ -0,0 +1,235 @@
+      },
+      { status: error.status },
+    )
+  }
+
+  if (error) {
+    logger.error({ error }, 'Service Providers Signup: Error')
+    return NextResponse.json(
+      { success: false, message: error.message, error },
+      { status: 500 },
+    )
+  }
+
+  return NextResponse.json({ success: true })
+}
+
+/**
+ * Handles the signup form for our Service Provider users.
+ */
+async function handleServiceProviderSignup(
+  req: Request,
+): Promise<
+  | { success: false; error: Error | PostgrestError }
+  | { success: true; error: null }
+> {
+  // validation
+  const supabase = createClient()
+
+  const validationResult = serviceProviderSignupFormSchema.safeParse(
+    await req.formData(),
+  )
+
+  if (validationResult.success === false) {
+    return { success: false, error: validationResult.error }
+  }
+
+  const {
+    email,
+    password,
+    company_name: companyName,
+    abn,
+    acn,
+    industry,
+    preferred_name: preferredName,
+    fullname,
+  } = validationResult.data
+
+  // sign up new user
+  const metadata = {
+    role: 'service-provider',
+    onboarded: false,
+    preferred_name: preferredName,
+    fullname,
+    service_providers: {
+      stripe_account_id: null,
+      onboarding_url: null,
+    },
+    companyDetails: {
+      companyName,
+      abn,
+      acn,
+      industry,
+    },
+  }
+
+  const signupResult = await supabase.auth.signUp({
+    email,
+    password,
+    options: {
+      data: metadata,
+    },
+  })
+
+  if (signupResult.error) {
+    return { success: false, error: signupResult.error }
+  }
+
+  const userId = signupResult.data?.user?.id as string
+
+  // new service provider
+  const insertServiceProviderResult = await supabase
+    .from('service_providers')
+    .insert([
+      {
+        user_id: userId,
+        company_name: companyName,
+        abn,
+        acn,
+        industry,
+        preferred_name: preferredName,
+        fullname,
+        slug: slugify(companyName),
+      },
+    ])
+
+  if (insertServiceProviderResult.error) {
+    return {
+      success: false,
+      error: insertServiceProviderResult.error,
+    }
+  }
+
+  // create new stripe account
+  const [stripeAccountError, stripeAccount] = await to(
+    createStripeConnectAccount({
+      userId,
+      email,
+    }),
+  )
+
+  if (stripeAccountError) {
+    return { success: false, error: stripeAccountError }
+  }
+
+  const serviceProviderId = stripeAccount.id
+
+  // get onboarding url
+  const [stripeRedirectError, stripeRedirect] = await to(
+    createStripeConnectRedirectLink(serviceProviderId),
+  )
+
+  if (stripeRedirectError) {
+    return { success: false, error: stripeRedirectError }
+  }
+
+  // save to metadata
+  const updatedMetadata = {
+    role: 'service-provider',
+    onboarded: false,
+    preferred_name: preferredName,
+    fullname,
+    service_providers: {
+      stripe_account_id: serviceProviderId,
+      onboarding_url: stripeRedirect.url,
+    },
```

Comment:
I think this should be a database table instead of metadata. Having an onboarding status table would make it easier to audit by an admin. 

I won't follow through with building an admin dashboard for, but I'd like to keep that door open. Will look into adding a new table.